sandboxed_api/tools/clang_generator/emitter.cc

```diff
@@ -187,8 +188,42 @@ std::string PrintAstDecl(const clang::Decl* decl) {
   //   - Rewrite function pointers using std::add_pointer_t<>;
 
   if (const auto* record = llvm::dyn_cast<clang::CXXRecordDecl>(decl)) {
+    std::string template_textual_params;
+    if (const auto* template_inst_decl =
```

Comment:
probably worth extracting it into a function
those deeply nested if-s would probably just be early returns then.

Using `absl::StrJoin` seems clearer.
This part is most probably not performance critical and temporary strings are created anyhow (e.g. `std::string(template_param->getName())` - this could probably be just a `absl::string_view`, but just going with `absl::StrJoin` is clearer).

Reply:
Done.

Regarding `std::string(template_param->getName())`, the `getName()` method returns an `llvm::StringRef` (like an `absl::string_view`), so we need the conversion to `std::string`.